gravitee-am-service/src/test/java/io/gravitee/am/service/CertificateServiceTest.java

```diff
@@ -195,76 +245,155 @@ public void shouldDelete_certificateWithClients() {
+    private TestObserver<Certificate> defaultCertificate(int keySize, String algorithm, boolean shouldBeSuccessful) throws Exception {
+        initializeCertificatSettings(keySize, algorithm);
+
+        when(certificateRepository.create(any())).thenReturn(Single.just(new Certificate()));
+        when(eventService.create(any(Event.class))).thenReturn(Single.just(new Event()));
+
+        CertificateSchema schema = new CertificateSchema();
+        schema.setProperties(Collections.emptyMap());
+        doReturn(schema).when(objectMapper).readValue(anyString(), eq(CertificateSchema.class));
+        doReturn(mock(ObjectNode.class)).when(objectMapper).createObjectNode();
+        doReturn("certificate").when(objectMapper).writeValueAsString(any(ObjectNode.class));
+        doReturn(mock(JsonNode.class)).when(objectMapper).readTree(eq("certificate"));
+
+        when(certificatePluginService.getSchema(CertificateServiceImpl.DEFAULT_CERTIFICATE_PLUGIN))
+                .thenReturn(Maybe.just(CERTIFICATE_SCHEMA_DEFINITION));
+        TestObserver<Certificate> testObserver = certificateService.create(DOMAIN_NAME).test();
+        testObserver.awaitTerminalEvent();
+
+        if (shouldBeSuccessful) {
+            verify(certificateRepository).create(argThat(cert -> cert.isSystem()
+                    && cert.getDomain().equals(DOMAIN_NAME)
+                    && cert.getName().equals("Default")
+            ));
+        }
+
+        return testObserver;
+    }
+
+    private void initializeCertificatSettings(int keySize, String algorithm) {
         when(environment.getProperty(eq("domains.certificates.default.keysize"), any(), any())).thenReturn(keySize);
         when(environment.getProperty(eq("domains.certificates.default.validity"), any(), any())).thenReturn(365);
         when(environment.getProperty(eq("domains.certificates.default.name"), any(), any())).thenReturn("cn=Gravitee.io");
         when(environment.getProperty(eq("domains.certificates.default.algorithm"), any(), any())).thenReturn(algorithm);
         when(environment.getProperty(eq("domains.certificates.default.alias"), any(), any())).thenReturn("default");
         when(environment.getProperty(eq("domains.certificates.default.keypass"), any(), any())).thenReturn("gravitee");
         when(environment.getProperty(eq("domains.certificates.default.storepass"), any(), any())).thenReturn("gravitee");
+    }
+
+    @Test
+    public void shouldRotate_defaultCertificate_Rsa() throws Exception {
+        final var now = LocalDateTime.now();
+        final Certificate certOldest = new Certificate();
+        certOldest.setSystem(true);
+        certOldest.setDomain(DOMAIN);
+        certOldest.setName("Cert-1");
+        certOldest.setConfiguration("{\"content\":\"03c14ae4-5f9b-4636-814a-e45f9bb636c4.p12\",\"storepass\":\"gravitee\",\"alias\":\"default\",\"keypass\":\"gravitee\"}");
```

Comment:
Again, hard coding json (or any other) syntax as a string leads to very hard to maintain code, even for tests. I would prefer to see these being read from resource files.